[DOCS] Option to schedule Osquery packs for individual policies or globally (#146482)
elastic/kibana: docs/osquery/osquery.asciidoc

. Click the **Packs** tab.
. Click **Add pack** to create a new pack, or click the name of an existing pack, then **Edit** to add queries to an existing pack.
. Provide a name for the pack. The short description is optional. 
. Schedule the pack to be deployed on specified agent policies (*Policy*) or on all agent policies (*Global*).
+
If you choose the *Policy* option, you must also configure one or both of the following fields: 
+
NOTE: When defining pack deployment details, you cannot configure the same policy multiple times. In other words, after specifying a policy, you can either choose to deploy the pack to all of the policy's agents or only a subset. You cannot choose both.

** *Scheduled {agent} policies (optional)*: Allows you to deploy the pack to specific agent policies. By default, the pack is deployed to all {agents} that are registered to the policies you define.  

> ** *Partial deployment (shards)*: Allows you to partially deploy the pack to specified agent policies. After defining a policy, use the *Shard* slider to limit the number of agents to which the pack is deployed. For example, after specifying a policy, you can choose to deploy the pack to half of the policy's agents by selecting 50% on the slider.

Reviewer: 'to limit the number of agents' - what do you think about changing number (I understand it as quantity) into something emphasizing '%' ?

Author: Sure, @tomsonpl! To make sure I'm understanding you correctly, are you asking that the word "number" be replaced with something less definite, such as "portion", "amount", or "subset"? For example:

 ```suggestion
** *Partial deployment (shards)*: Allows you to partially deploy the pack to specified agent policies. After defining a policy, use the *Shard* slider to set the amount of agents to which the pack is deployed. For example, after specifying a policy, you can choose to deploy the pack to half of the policy's agents by selecting 50% on the slider.
```

Reviewer: Yes @nastasha-solomon. that was exactly my point, thank you:)  Do you think it makes sense?

Author: Yeah, I think using less definite terms to describe the sharding feature is a good call since users can't actually pick which agents will run the pack. They can only specify a percentage of the group.

. If you are creating a new pack, add queries to schedule:

 ** Click *Add query* and then add a saved query or enter a new query.
Each query must include a unique query ID and the interval at which it should run.
Optionally, set the minimum Osquery version and platform,
or <<osquery-map-fields,map ECS fields>>. When you add a saved query to a pack, this adds a copy of the query. A connection is not maintained between saved queries and packs.

> ** Upload queries from a `.conf` query pack by dragging the pack to the drop zone under the query table. To explore the community packs that Osquery publishes, click *Example packs*.

Reviewer: Speaking of config, do you think it might be worth to mention that the policy's agents % is specified in config as 'shard' ?

Author: Yes, good reminder! I'll mention that under step 4 in this section and then again under step 5 [here](https://www.elastic.co/guide/en/kibana/current/manage-osquery-integration.html#osquery-custom-config).

. Click the **Packs** tab.
. Click **Add pack** to create a new pack, or click the name of an existing pack, then **Edit** to add queries to an existing pack.
. Provide a name for the pack. The short description is optional. 
. Schedule the pack to be deployed on specified agent policies (*Policy*) or on all agent policies (*Global*).
+

> TIP: Pack deployment details are stored within the <<osquery-custom-config,Osquery configuration>>. The `shard` field value shows the percentage of agents within the policy to which the pack has been deployed.  

Author: This is better and easier to read. Sometimes I forget that the clearer option is to describe what something is, rather than what something shows or appears to be.

+
If you choose the *Policy* option, you must also configure one or both of the following fields: 
+

> NOTE: When defining pack deployment details, you cannot configure the same policy multiple times. In other words, after specifying a policy, you can either choose to deploy the pack to all of the policy's agents or only a subset. You cannot choose both.

Reviewer 2: This note interrupts the flow from the lead-in above and the options below -- is there some other place it could go?

Author: Added to the previous line. Lemme know if it's still not flowing and I'll reiterate

Reviewer 2: @nastasha-solomon I think the note is still in the same place? From the latest diff it looks like it's still between the lead-in "configure these fields" and the two bulleted options.

** *Scheduled {agent} policies (optional)*: Allows you to deploy the pack to specific agent policies. By default, the pack is deployed to all {agents} that are registered to the policies you define.  

> ** *Partial deployment (shards)*: Allows you to partially deploy the pack to specified agent policies. After defining a policy, use the *Shard* slider to set the amount of agents to which the pack is deployed. For example, after specifying a policy, you can choose to deploy the pack to half of the policy's agents by selecting 50% on the slider.

Reviewer 2: I had trouble understanding what "partially deploy" means, which might just be my own inexperience with this feature. But the first sentence made me think that "partial" meant "some of the policies" when I think it should be "some of the agents on each policy"?
```suggestion
** *Partial deployment (shards)*: Allows you to deploy the pack to a portion of the agents on each specified agent policy. After defining a policy, use the *Shard* slider to set the amount of agents to which the pack is deployed. For example, after specifying a policy, you can choose to deploy the pack to half of the policy's agents by selecting 50% on the slider.
```

Author: Ah, I see what you're saying and your revision is more accurate. This setting allows users to deploy Osquery to some of the agents on a specified agent policy.

> . If you are creating a new pack, add queries to schedule:

Reviewer 2: ```suggestion
. If you're creating a new pack, add queries to schedule:
```
